Comment notifications cleanup (#476 ish)

TLSM opened the pull request:
The story of this PR is that I set out to address #476, hard wrapped some lines to make it easier to reason about, changed the notification subpages to use real paths rather than query string parameters, and realized that we probably mostly fixed the issue with one of the many comment filtering / moderation state / comment_on_publish reworks.

Our notifications logic is nearly identical to upstream differing mainly in: upstream sorts comment replies by new, whereas ours was by old. This has been remedied (as [suggested in the meta thread at the time](https://www.themotte.org/post/216/meta-something-shiny-and-two-things/56269?context=8#context)).

Given that the issue doesn't exist on upstream, my theory is it's because of fork-specific comment filtering code. A reply can still get quasi-buried through the following sequence: 1) a reply is filtered, 2) a different unfiltered reply is made, 3) the first reply is then unfiltered. These will appear sorted by submission time descending, not by first time visible descending, which may be perceived as burial, though the notification bell will still light up. We don't have the data readily available in the schema to fix this with the context-included approach, though sort-by-new may ameliorate this in practice. 

NB: if we wanted to go to reddit-style comment notifications (just the reply, no context), it would be trivial to do so and _would_ truly solve the aforementioned scenario where initially-filtered comments can be buried. Am willing to amend PR if this is requested.

tl;dr— actual function differences: `/notifications?messages=true` → `/notifications/messages` (etc), comment replies in notifs are sorted by new. Code broken up into more readable blocks.

TLSM committed:
Remove unused reddit mentions notifications

In the distant past, the codebase would check pushshift for keyword
mentions and generate notifications to admins about them. I can't
find where we removed it, but I don't think it was ever operative.

Since these notifications don't get generated, we don't need a
notifications page for them.

TLSM committed:
Split notification routes, hard wrap, sort order

Change the notifications subpages to be distinct route handlers with
actual paths rather than query parameters. They already were a massive
conditional chain with almost no common logic.

Hard wrap some of the more egregious query lines. Use less duplicated
code for shadowban exclusion.

Only major functionality change which is somewhat related to #476 is
to sort subtrees by Comment.id DESC. Otherwise, upstream is
substantially the same as TheMotte. Given that upstream didn't
experience #476, I think the issue may have been resolved by prior
changes to filtering / comment visibility & moderation.

zorbathut commented:
> NB: if we wanted to go to reddit-style comment notifications (just the reply, no context), it would be trivial to do so

Yeah, this is probably the right choice - the current behavior is, honestly, weird. I was expecting it to be more complicated, if it's easy, go for it!

TLSM committed:
Use reddit-style notifs without context

The notifications schema is already set up for reddit-style
chronological notifications. We simply have to remove the logic that
builds the context and patch up a few places in the frontend that
were expecting full reply trees.

`Comment.header_msg` previously expected the user's own comment to
be top-level in comment replies. Logic is revised to expect the actual
reply.

`files.routes.front.notifications_main` now has reduced query volume
because we aren't expiring the session when marking notifications read.
This also allows us to remove the unused `comms` variable, which is a
makeshift "pattern" of storing duplicate database replies across a
commit so the templates don't requery (thus losing data attached to
the runtime object, like `c.unread` / `c.notif_utc`).

We move the `is_notification_page` flag to the route callers rather
than templates checking `request.path`.

Minor UI style: "Clear all notifications" -> "Mark All Read", since
this was a persistent point of user confusion upstream, with people
expecting their inbox to empty out. Also less margin between notifs
to be consistent with tighter comment display elsewhere and removed
need to separate groups of comments vs single comments.
